pyproject.toml

```diff
@@ -13,3 +13,16 @@ dependencies = [
 [project.scripts]
 helloworld = "hello_package.helloworld:main"
 numpydemo = "hello_package.example:main"
+
+[project.optional-dependencies]
+dev = [
+    "mypy",
+    "isort",
+    "flake8",
+    "flake8-pyproject",
+    "black",
+] # to remove and add pre commit
```

Comment:
Just found some answers on pyproject.toml and requirements.txt - https://stackoverflow.com/questions/74508024/is-requirements-txt-still-needed-when-using-pyproject-toml 

Perhaps it is useful...

Reply:
cool, but I do both so that pip install doesn't try to install hard dependencies again